src/api/namespaces/atomicmarket/handlers/sales2.get-sales-templates.test.ts

```diff
@@ -0,0 +1,423 @@
+import 'mocha';
+import {expect} from 'chai';
+import {RequestValues} from '../../utils';
+import {initAtomicMarketTest} from '../test';
+import {getTestContext} from '../../../../utils/test';
+import {getSalesTemplatesV2Action} from './sales2';
+
+const {client, txit} = initAtomicMarketTest();
+
+async function getSalesIds(values: RequestValues): Promise<Array<number>> {
+    const testContext = getTestContext(client);
+
+    await client.query('SELECT update_atomicmarket_sales_filters()');
+
+    const result = await getSalesTemplatesV2Action({
+        symbol: 'TEST',
+        ...values,
+    }, testContext);
+
+    return result.map((s: any) => s.sale_id);
+}
+
+async function initTest(saleValues: Record<string, any> = {}): Promise<{template_id: number, sale_id: number}> {
+    const {template_id} = await client.createTemplate();
+
+    const {offer_id} = await client.createOfferAsset({}, {template_id});
+    const {sale_id} = await client.createSale({
+        offer_id,
+        ...saleValues,
+    });
+
+    return {
+        sale_id,
+        template_id: (await client.createTemplate()).template_id,
+    };
+}
+
+describe('AtomicMarket Sales API', () => {
+
+    describe('getSalesTemplatesAction', () => {
+
+        txit('filters by settlement symbol', async () => {
+            const {template_id} = await initTest();
+            await client.createToken({token_symbol: 'TOKEN1'});
+            const {offer_id} = await client.createOfferAsset({}, {template_id});
+            const {sale_id} = await client.createSale({offer_id, settlement_symbol: 'TOKEN1'});
+
+            expect(await getSalesIds({symbol: 'TOKEN1', burned: false}))
+                .to.deep.equal([sale_id]);
+        });
+
+        txit('filters by minimum price', async () => {
+            const {template_id} = await initTest({
+                listing_price: 100000000,
+            });
+
+            const {offer_id} = await client.createOfferAsset({}, {template_id});
+            const {sale_id} = await client.createSale({
+                offer_id,
+                listing_price: 200000000,
+            });
+
+            expect(await getSalesIds({min_price: '2', burned: false}))
```

Comment:
Let's test also the reverse, if the min price is above the listing price, this should not be returned

Reply:
Or well I dont understand correctly, why min price  = 2 and the two of those are not returned?

Reply:
100000000 = 1
200000000 = 2

https://github.com/pinknetworkx/eosio-contract-api/blob/0b5280c9e191d520de38a7ea37dea03391edcae5/src/api/namespaces/atomicmarket/handlers/sales2.ts#L153